.github/workflows/check_clean_branch.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+# Exits if any uncommitted changes are found.
+
+set -o errexit
```

Comment:
can you add comments to this file explaining the steps

Reply:
added comments.